relay.go

```diff
@@ -52,166 +61,271 @@ func NewRelay(ctx context.Context, h host.Host, opts ...RelayOpt) (*Relay, error
 		incoming: make(chan *Conn),
 	}
 
-	h.SetStreamHandler(StopID, r.HandleNewStopStream)
-
 	for _, opt := range opts {
 		switch opt {
 		case OptActive:
 			r.active = true
 		case OptHop:
-			h.SetStreamHandler(HopID, r.HandleNewHopStream)
+			r.hop = true
 		default:
 			return nil, fmt.Errorf("unrecognized option: %d", opt)
 		}
 	}
 
+	h.SetStreamHandler(ProtoID, r.handleNewStream)
+
 	return r, nil
 }
 
-func (r *Relay) Dial(ctx context.Context, relay peer.ID, dest ma.Multiaddr) (*Conn, error) {
-	s, err := r.host.NewStream(ctx, relay, HopID)
+func (r *Relay) Dial(ctx context.Context, relay pstore.PeerInfo, dest pstore.PeerInfo) (*Conn, error) {
+	err := r.host.Connect(ctx, relay)
 	if err != nil {
 		return nil, err
 	}
 
-	if err := writeLpMultiaddr(s, dest); err != nil {
+	s, err := r.host.NewStream(ctx, relay.ID, ProtoID)
```

Comment:
Not really. It's actually necessary to prevent some potential race conditions (open a connection, connection gets garbage collected (once we start actually doing that), try to open a stream). It's also just really convenient.